persist/fs/commitlog/reader.go

```diff
@@ -22,27 +22,32 @@ package commitlog
 
 import (
 	"bufio"
+	"context"
 	"encoding/binary"
 	"errors"
+	"io"
 	"os"
+	"sync"
 	"time"
 
 	"github.com/m3db/m3db/digest"
 	"github.com/m3db/m3db/persist/encoding"
 	"github.com/m3db/m3db/persist/encoding/msgpack"
 	"github.com/m3db/m3db/persist/schema"
 	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3x/checked"
 	"github.com/m3db/m3x/pool"
 	xtime "github.com/m3db/m3x/time"
 )
 
+const decoderInBufChanSize = 100
```

Comment:
Would it be better to make this an `Options` parameter?

Reply:
Honestly I'm not sure. I just kind of made up a number that seemed large enough. Changing it shouldn't have too much of an effect on things since all the workers should be doing almost exactly the same amount of work and it seems a little too fine-grained for people to be tuning manually

Reply:
Could you verify what impact it has in a couple of runs, can decide whether we should move it up after.

Reply:
any word on this

Reply:
yep I messed around with it when I was debugging that deadlock issue and the code is still correct even if I set the value to 1, but I didn't notice any perf difference between 100 and 1000 so I think this is an ok value for now

Reply:
Dope, sounds good